### PR TITLE
Modular clustering

### PR DESCRIPTION
**Pull request by ttngu207**
Separation of the `Clustering.make()` into multiple smaller steps: preprocessing, kilosort, postprocessing. 
This is structured in the new module `ecephys_spike_sorting` - a DataJoint pipeline design specific to the spike sorting routine from https://github.com/jenniferColonell/ecephys_spike_sorting

The new `ecephys_spike_sorting` DataJoint schema is designed not as a redesign/replacement to the current implementation of the `Clustering`, but as a supplementing component, that can be imported, activated, and used together with the existing ephys pipeline

### CODE REVIEW

File: element_array_ephys/spike_sorting/ecephys_spike_sorting.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+                KS2ver=f'{Decimal(clustering_method.replace("kilosort", "")):.1f}',
+                run_CatGT=True,
+            )
+            run_kilosort._modules = ["kilosort_helper"]
+            run_kilosort._CatGT_finished = True
+            run_kilosort.run_modules()
+        elif acq_software == "Open Ephys":
+            oe_probe = ephys.get_openephys_probe_data(key)
+
+            assert len(oe_probe.recording_info["recording_files"]) == 1
```
Comment (CBroz1) [inline element_array_ephys/spike_sorting/ecephys_spike_sorting.py:215]: Do some recordings split across multiple files? If so, maybe we can throw a NotImplementedError here saying we can't handle truncated files? Maybe suggesting a tool for concatenation?

File: element_array_ephys/spike_sorting/ecephys_spike_sorting.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        output_dir = (ephys.ClusteringTask & key).fetch1("clustering_output_dir")
+        kilosort_dir = find_full_path(ephys.get_ephys_root_data_dir(), output_dir)
+
+        acq_software, clustering_method = (
+            ephys.ClusteringTask * ephys.EphysRecording * ephys.ClusteringParamSet & key
+        ).fetch1("acq_software", "clustering_method")
+        assert clustering_method in (
+            "kilosort2",
+            "kilosort2.5",
+            "kilosort3",
```
Comment (CBroz1) [inline element_array_ephys/spike_sorting/ecephys_spike_sorting.py:263]: Given that this tuple is reused, should it migrate elsewhere as a SUPPORTED_KILOSORT_VERSIONS constant?

**Review by CBroz1 — COMMENTED** [review]
Thanks for your work here @ttngu207!

I added some comments as suggestions before merge, but others for my own understanding. I hope they're helpful

File: element_array_ephys/readers/kilosort_triggering.py [hunk 3]
```diff
@@ -759,8 +767,8 @@ def _write_channel_map_file(
     )
 
     if is_0_based:
-        channel_ind += 1
-        shank_ind += 1
+        channel_ind = channel_ind + 1
```
[the remark this answers is not in the document]
Comment (CBroz1) [reply element_array_ephys/readers/kilosort_triggering.py:770]: After discussion, motivated by permission error

**Comment by ttngu207** [comment]
To be replaced with a new PR to `modular_clustering` branch

**Pull request closed without merging**